File: fix_views.py
import os

def update_view(filepath, header, footer):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find start of main content
    start = content.find('<div class="d-flex justify-content-between')
    if start == -1: start = content.find('<div class="page-header')
    if start == -1: start = content.find('<div class="card-custom">')

    # Find end of main content
    end = content.find('</body>')
    if end == -1: end = len(content)
    else:
        end = content.rfind('</div>', 0, end)
        end = content.rfind('</div>', 0, end)

    main_content = content[start:end]

    # Fix actions
    main_content = main_content.replace('admin/produk/simpan', 'petani/produk/simpan')
    main_content = main_content.replace('admin/produk/update', 'petani/produk/update')
    main_content = main_content.replace('admin/produk', 'petani/produk')

    # If it's edit or tambah, make sure it has enctype and the gallery input
    if 'tambah' in filepath or 'edit' in filepath:
        if 'enctype="multipart/form-data"' not in main_content:
            main_content = main_content.replace('method="post"', 'method="post" enctype="multipart/form-data"')

        # Add foto_utama if not exists (for edit)
        if 'name="foto_utama"' not in main_content:
            img_input = '''
                            <div class="form-group mb-3">
                                <label class="font-weight-bold">Foto Utama</label>
                                <input type="file" name="foto_utama" class="form-control" accept="image/*">
                            </div>
'''
            main_content = main_content.replace('<div class="form-group mb-3">\n                                <label class="font-weight-bold">Status Produk', img_input + '<div class="form-group mb-3">\n                                <label class="font-weight-bold">Status Produk')

        # Add gallery input
        if 'name="galeri[]"' not in main_content:
            gal_input = '''
                            <div class="form-group mb-3">
                                <label class="font-weight-bold">Galeri Tambahan (Bisa pilih banyak)</label>
                                <input type="file" name="galeri[]" multiple class="form-control" accept="image/*">
                            </div>
'''
            main_content = main_content.replace('<div class="form-group mb-3">\n                                <label class="font-weight-bold">Status Produk', gal_input + '<div class="form-group mb-3">\n                                <label class="font-weight-bold">Status Produk')

    new_content = header + '\n<div class="main-content">\n' + main_content + '\n' + footer
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("Updated", filepath)

base_path = 'C:/Users/USER/.gemini/antigravity/scratch/poktan_kopi/'
with open(base_path + 'application/views/petani/v_produk.php', 'r', encoding='utf-8') as f:
    v_produk = f.read()

header_pos = v_produk.find('<!-- MAIN CONTENT -->')
header = v_produk[:header_pos]

footer_pos = v_produk.find('<script', header_pos)
footer = '</div>\n</div>\n' + v_produk[footer_pos:]

update_view(base_path + 'application/views/petani/produk_tambah.php', header, footer)
update_view(base_path + 'application/views/petani/produk_edit.php', header, footer)
update_view(base_path + 'application/views/petani/produk_detail.php', header, footer)

File: fix_views.php
<?php
$v_produk = file_get_contents('application/views/petani/v_produk.php');

$header_pos = strpos($v_produk, '<!-- MAIN CONTENT -->');
$header = substr($v_produk, 0, $header_pos);

$footer_pos = strpos($v_produk, '<script', $header_pos);
$footer = "</div>\n</div>\n" . substr($v_produk, $footer_pos);

function update_view($file, $header, $footer) {
    $content = file_get_contents($file);
    
    // Extract the form/main content from the target file
    // For produk_tambah, it's inside <div class="card-custom"> or <div id="content">
    $start = strpos($content, '<div class="d-flex justify-content-between');
    if ($start === false) $start = strpos($content, '<div class="page-header');
    if ($start === false) $start = strpos($content, '<div class="card-custom">');
    
    // find the end (usually before <script> or </body>)
    $end = strpos($content, '</body>');
    if ($end === false) $end = strlen($content);
    else {
        // Find the closing div of content
        $end = strrpos(substr($content, 0, $end), '</div>');
        $end = strrpos(substr($content, 0, $end), '</div>');
    }
    
    $main_content = substr($content, $start, $end - $start);
    
    // In produk_tambah, form action is wrong: admin/produk/simpan -> petani/produk/simpan
    $main_content = str_replace('admin/produk/simpan', 'petani/produk/simpan', $main_content);
    $main_content = str_replace('admin/produk/update', 'petani/produk/update', $main_content);
    $main_content = str_replace('admin/produk', 'petani/produk', $main_content);

    // If it doesn't have enctype for edit, we might need to add it later.
    
    $new_content = $header . "\n<div class=\"main-content\">\n" . $main_content . "\n" . $footer;
    file_put_contents($file, $new_content);
    echo "Updated $file\n";
}

update_view('application/views/petani/produk_tambah.php', $header, $footer);
update_view('application/views/petani/produk_edit.php', $header, $footer);
update_view('application/views/petani/produk_detail.php', $header, $footer);

?>
